pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -199,6 +235,40 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}, nil
 }
 
+func patchSecret(ctx context.Context, c client.Client, scheme *runtime.Scheme, secret *v1.Secret, mutationFunc func() error) error {
+	err := c.Get(ctx, client.ObjectKeyFromObject(secret), secret.DeepCopy())
```

Comment:
this might look weird, but we just wanna see if we get an 404. We do not want to pass in the `*v1.Secret` as it would be mutated.